pkg/ccl/backupccl/backup_test.go

```diff
@@ -9191,3 +9192,115 @@ func TestBackupRestoreSeparateIncrementalPrefix(t *testing.T) {
 		sqlDB.Exec(t, "DROP DATABASE inc_fkdb;")
 	}
 }
+
+func TestEphemeralBackupAndRestore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	_, sqlDB, iodir, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, 10,
+		InitManualReplication, base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(), // speeds up test
+					SpanConfig: &spanconfig.TestingKnobs{
+						SQLWatcherCheckpointNoopsEveryDurationOverride: 100 * time.Millisecond,
+					},
+				},
+			},
+		})
+	defer cleanupFn()
+
+	_, restoreDB, cleanup := backupRestoreTestSetupEmpty(t, singleNode, iodir, InitManualReplication,
+		base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(), // speeds up test
+				},
+			},
+		})
+	defer cleanup()
+	restoreDB.Exec(t, `CREATE DATABASE data`)
+
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.rangefeed.enabled = true`)
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`)
+
+	sqlDB.Exec(t, `CREATE TABLE data.foo (id INT, INDEX bar(id))`)
+	sqlDB.Exec(t, `INSERT INTO data.foo select * from generate_series(1,10)`)
+
+	checkNumRestoredRows := func(backupDir string, retryAttempt, expectedRows int) error {
+		backupPath := fmt.Sprintf("%s/%d", backupDir, retryAttempt)
+		defer func() {
+			restoreDB.Exec(t, `DROP TABLE IF EXISTS data.foo`)
+		}()
+
+		sqlDB.Exec(t, `BACKUP TABLE data.foo TO $1`, backupPath)
+		restoreDB.Exec(t, `RESTORE TABLE data.foo FROM $1`, backupPath)
+
+		res := restoreDB.QueryStr(t, `SELECT count(*) FROM data.foo`)
+		numRows, err := strconv.Atoi(res[0][0])
+		require.NoError(t, err)
+		if numRows != expectedRows {
+			return errors.Newf("expected %d rows, but found %d", expectedRows, numRows)
+		}
+		return nil
+	}
+
+	// Set table to ephemeral and back it up. The ExportRequest should be a noop
+	// and backup no data.
+	sqlDB.Exec(t, `ALTER TABLE data.foo SET EPHEMERAL DATA`)
+
+	var retryAttempt int
+	testutils.SucceedsSoon(t, func() error {
```

Comment:
I might rework this test to actually fetch the replica and wait for the Ephemeral bit to be set, instead of wrapping all statements in a SucceedsSoon.